### PR TITLE
Feature: vacancies stages board

### PR DESCRIPTION
**Pull request by R-Mielamud**

### CODE REVIEW

**Comment by anton-matsyshyn** [comment]
Please, add separate routing module to vacancies module and use it for declaring vacancies navigation

**Comment by anton-matsyshyn** [comment]
A lot unnecessary data is loaded from GetByVacancyAsyncId endpoint. We need to load list of stages, each stage with list of candidate. Create dto for that puprose

**Comment by anton-matsyshyn** [comment]
GetByVacancyAsync may be optimized. Check next overload of QueryAsync:

```
connection.QueryAsync<Stage, VacancyCandidate[], Stage>(sql, (stage, candidates) =>
            {
                stage.Candidates = candidates;
                return stage;
            });
```

File: frontend/ATS/src/app/vacancies/components/candidate-card/candidate-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Component, Input, OnInit } from '@angular/core';
+import moment from 'moment';
+
+// This line can't be shorter
+// eslint-disable-next-line max-len
+import { VacancyCandidateWithApplicant } from 'src/app/shared/models/vacancy-candidates/with-applicant';
```
Comment (Yaroslaww-1) [inline frontend/ATS/src/app/vacancies/components/candidate-card/candidate-card.component.ts:6]: What about
```
import {
   VacancyCandidateWithApplicant
} from 'src/app/shared/models/vacancy-candidates/with-applicant';
```
?

**Comment by R-Mielamud** [comment]
> GetByVacancyAsync may be optimized. Check next overload of QueryAsync:
> 
> ```
> connection.QueryAsync<Stage, VacancyCandidate[], Stage>(sql, (stage, candidates) =>
>             {
>                 stage.Candidates = candidates;
>                 return stage;
>             });
> ```

This multi-mapping is a very tricky thing, and it can't be used here. 3 queries instead of 4 aren't much faster :)

File: frontend/ATS/src/app/vacancies/components/candidate-card/candidate-card.component.ts (hunk 1 above)
Comment (R-Mielamud) [reply frontend/ATS/src/app/vacancies/components/candidate-card/candidate-card.component.ts:6]: VSCode doesn't allow this

**Comment by anton-matsyshyn** [comment]
Please, examine [this](https://dapper-tutorial.net/result-multi-mapping#example-query-multi-mapping-one-to-many) approach. We can't afford such multiple queries

**Review by anton-matsyshyn — APPROVED** [review]

**Pull request merged**